main.tf

```diff
@@ -0,0 +1,141 @@
+resource "aws_s3_bucket" "data" {
+  # bucket is public
+  # bucket is not encrypted
+  # bucket does not have access logs
+  # bucket does not have versioning
+  bucket        = "${local.resource_prefix.value}-data"
+  force_destroy = true
+  tags = merge({
+    Name        = "${local.resource_prefix.value}-data"
+    Environment = local.resource_prefix.value
+    }, {
+    git_commit           = "4d57f83ca4d3a78a44fb36d1dcf0d23983fa44f5"
+    git_file             = "terraform/aws/s3.tf"
+    git_last_modified_at = "2022-05-18 07:08:06"
+    git_last_modified_by = "nimrod@bridgecrew.io"
+    git_modifiers        = "34870196+LironElbaz/nimrod/nimrodkor"
+    git_org              = "bridgecrewio"
+    git_repo             = "terragoat"
+    yor_trace            = "0874007d-903a-4b4c-945f-c9c233e13243"
+  })
+}
+
+resource "aws_s3_bucket_object" "data_object" {
```

Comment:
<details><summary>
<img src="https://bc-network-graph-icons.s3.us-west-2.amazonaws.com/prCommentsSeverityIcons/low_new.svg" alt="LOW" width="24px" align="center">&nbsp;&nbsp;<b style="color:black" >Ensure S3 bucket Object is encrypted by KMS using a customer managed Key (CMK)</b>
            <br>
            &nbsp;&nbsp;&nbsp;&nbsp;Resource: <a href="https://github.com/marynaKK/terragoat/blob/marynaKK-patch-1/main.tf#L23-L40">aws_s3_bucket_object.data_object</a> | Bridgecrew ID: <code>BC_AWS_GENERAL_106</code> | Checkov ID: <a href="https://docs.prismacloud.io/en/enterprise-edition/policy-reference/aws-policies/aws-general-policies/bc-aws-general-106">CKV_AWS_186</a>
            <br>
            
</summary>
<h4>How to Fix</h4>

```tf
resource "aws_s3_bucket_object" "object" {
  bucket = "your_bucket_name"
  key    = "new_object_key"
  source = "path/to/file"
+ kms_key_id = "ckv_kms"

  # The filemd5() function is available in Terraform 0.11.12 and later
  # For Terraform 0.11.11 and earlier, use the md5() function and the file() function:
  # etag = "${md5(file("path/to/file"))}"
  etag = filemd5("path/to/file")
}
```

<h4>Description</h4>


This is a simple check to ensure that the S3 bucket Object is using AWS key management - KMS to encrypt its contents.
To resolve add the ARN of your KMS or link on creation of the object.

</details>
<p><img src="https://bc-network-graph-icons.s3.us-west-2.amazonaws.com/prCommentsBotLogos/bc.svg" alt="Bridgecrew" width="18px">&nbsp;<small>Comment by Bridgecrew Bot</small></p>